### ISSUE TITLE
install problem on mac os 10.13.4

### ISSUE THREAD

**Issue by rwelty1889** [mapillary/mapillary_tools#234]
### Basic information
* Version: latest
* System: Mac OS 10.13.4

### Steps to reproduce behavior

 1. attempted install using python 2.7 (tried both apple installation and from mac ports)
 2. sudo -H pip install git+https://github.com/mapillary/mapillary_tools

### Expected behavior

successful installation of code

### Actual behavior

    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/private/tmp/pip-install-uewfe9_h/exifread/setup.py", line 5, in <module>
        from exifread import __version__
      File "/private/tmp/pip-install-uewfe9_h/exifread/exifread/__init__.py", line 2, in <module>
        from .classes import *
      File "/private/tmp/pip-install-uewfe9_h/exifread/exifread/classes.py", line 7, in <module>
        from .tags import *
      File "/private/tmp/pip-install-uewfe9_h/exifread/exifread/tags/__init__.py", line 3, in <module>
        import makernote
    ImportError: No module named 'makernote'

### Additional information

installation notes are confusing/inconsistent. article announcing the upgraded tool set lists upgrade to python 3 is impending but not yet done, but README says python>2.7 which means use python 3. which is it?

**Comment by rTeamWork** [comment]
I'm having the same issue on Windows.

**Comment by jernejaMislej** [comment]
Hi, thank you for reporting. We have corrected the Python version to be used for now, thank you for noting that.
`makernote` is not supposed to be an external module, there might be issues with the paths, can you try installing with `--no-cache-dir`

**Comment by rwelty1889** [comment]
tried --no-cache-dir with 2.7, no change in behavior. i'm going to be of limited help, i'm not a python wiz. but this note seems like it might be relevant; the makernote path may not always be automatically detected: https://github.com/ianare/exif-py/pull/25

**Comment by jernejaMislej** [comment]
hi,
thanks for the information, I will check if it is ok to use a newer version of exifread and let you know

**Comment by rwelty1889** [comment]
any status on this? i have a lot of imagery from my GoPro waiting on it.

**Comment by sandro1985** [comment]
I am having this issue as well on Windows 10 using python version 3.7.0

**Comment by jernejaMislej** [comment]
python version needs to be 2.7

**Comment by jernejaMislej** [comment]
@sandro1985  @rwelty1889  @rTeamWork I checked the compatibility of a newer version of `exifread` and all is ok, so the version of this dependency will be upgraded in the next release. 
Hopefully this will resolve the issues, if not please report here. Note that you still need to use Python 2.7

**Comment by jernejaMislej** [comment]
hi please try to install the latest release v0.1.0 and let us know if there are still issues with instalment

**Comment by rwelty1889** [comment]
will try it later today. thanks.

**Comment by rwelty1889** [comment]
resolved my own issue. in cases where multiple versions of python are installed, you use

pip-2.7 install mybarpackage

(for example) to force use of python 2.7

**Comment by ybkuang** [comment]
@rwelty1889 to authenticate, you can use the credential for you Mapillary user account.

**Comment by rwelty1889** [comment]
i tried those credentials. they were rejected.

the question still remains, what does failure to authenticate mean when running mapillary_tools process? i don't know if it succeeded or failed, or partially succeeded.

**Comment by rwelty1889** [comment]
to be very clear: if "mapillary_tools process" fails to authenticate, it continues to run while producing many, many error messages. i do not understand the state of my data, and i will not run this process again without clarification of this point.

**Comment by jernejaMislej** [comment]
Hi sorry for the late reply, I m on vacation until 10th August with limited access to internet. 
The process tool runs several sub-processing steps, user authentication is one of them. Even if one of the sub-processing steps fail, the rest will try to execute. User authentication is required for images to be uploaded ,but not to be geotagged and processed into sequences. 
You will not be able to upload the images without user authentication, even if the rest of the sub-processing steps were executed successfully. 
@Kaldea can you please help the user make sure the right credentials, ie username, email and pass are being used?

**Comment by Kaldea** [comment]
Hi @rwelty1889! Can you reach out to me with your credentials at support@mapillary.com for me to verify? Your username and email address is fine.

**Comment by rwelty1889** [comment]
i have updated my copy of mapillary_tools and worked through the issues. stuff is more-or-less working now although i think there still may be some uploaded but unprocessed images kicking about.

**Comment by rwelty1889** [comment]
for reference my mapillary account is nfgusedautoparts, email address rwelty@averillpark.net

**Comment by jernejaMislej** [comment]
@rwelty1889 thank you for getting back. We will make sure all your images get into the system.
If you have any more problems with the installation and authentication please open a new issue, closing this for now
